wallet/wallet-controller/src/lib.rs

```diff
@@ -39,34 +44,41 @@ const BLOCK_SYNC_ENABLED: bool = false;
 pub enum ControllerError<T: NodeInterface> {
     #[error("Node call error: {0}")]
     NodeCallError(T::Error),
+    #[error("Wallet file {0} error: {0}")]
+    WalletFileError(PathBuf, String),
+    #[error("Wallet error: {0}")]
+    WalletError(wallet::wallet::WalletError),
 }
 
 pub struct Controller<T: NodeInterface> {
+    chain_config: Arc<ChainConfig>,
+
     rpc_client: T,
 
-    wallet: DefaultWallet,
+    wds: Vec<WalletData<T>>,
```

Comment:
Why multiple wallets like this?

People who want to open multiple wallets should run the software multiple times. That simplifies things a lot.

Reply:
Even in the GUI, the handle for opening a wallet should be accessed multiple times to open multiple wallets.

For example, if there's a function `open_or_create_wallet`, for every wallet the user wants to open, that function is called, a controller is returned, and that controller is what's used in the GUI to communicate with the wallet.

Reply:
Okay, reverted.